### PR TITLE
Fix #63: Go to item page on click of PreviewItem

### LINKED ISSUE #63: Search results page should redirect to vessel and crew for those listings
Like issue #28, on the search results page, vessel and crew listings should be clickable and redirect to those pages.

### PR DESCRIPTION
**Pull request by lenmorld**
## Related Issue

Fixes #63 

## Checklist:

- [x] I have read the [contributor's guide](https://wildaid.github.io/contribute/index.html).
- [x] I linked an issue in the previous section
- [x] I have commented on the linked issue
- [x] I was assigned the linked issue (not required)
- [x] I have tested the change to the best of my ability against the [sandbox](https://wildaid.github.io/contribute/sandbox.html) or a [local build](https://wildaid.github.io/build).

Optional items:

- [ ] My change adds new text and requires a change to translations.
- [ ] My change requires a change to the documentation.
- [ ] I have submitted a PR to the [documentation repo](https://github.com/WildAid/wildaid.github.io).
- [ ] I was not able to test... (explain below, e.g. you did not have permissions to test a specific feature)
- [ ] This change depends O-FISH Realm repository changes (explain below)

* **Optional: Add any explanations here** 
Work done:
1. add shared, reusable functions in get-data to view items
2. use those functions in search preview item and crew view
3. add a pointer cursor to search preview item to make clickability apparent

* **Optional: Add any relevant screenshots here** 

ofish_issue_63

### CODE REVIEW

**Comment by lenmorld** [comment]
Preview of result:
ofish_issue_63

File: src/components/autofill/autofill.component.js [hunk 1]
```diff
@@ -7,37 +7,46 @@ import PreviewItem from "./preview-item/preview-item.component";
...
         />
         <PreviewItem
-          item={crew[0]}
+          item={crewItem}
           itemName={t("SEARCH.CREW_MEMBERS").toUpperCase()}
           icon="crew"
           previewName="name"
           subText={t("NAVIGATION.VESSELS")}
           searchWords={searchWords}
+          onClick={() => goCrewViewPage(crewItem)}
```
Comment (Sheeri) [inline src/components/autofill/autofill.component.js:40]: When I tested this out on the sandbox server, I searched for 'Jack' and got a match:
Screen Shot 2020-10-19 at 10 10 02 PM


I clicked on "Jack Harknass" under "Crew Members" and got this blank page:

Screen Shot 2020-10-19 at 10 10 09 PM

Note the URL: http://localhost:3000/#/crew/view/{%22captain.name%22:%22Jack%20Harknass%22}

Unfortunately, the crew view page depends on both the crew member's name and the license number of the crew member. Is there any way you can make sure the license number is retrieved on this page, and sent onto the next page?

**Review by Sheeri — CHANGES_REQUESTED** [review]
The redirect for vessel and boardings look great, I just had an issue with the crew redirect.

File: src/components/autofill/autofill.component.js (hunk 1 above)
Comment (lenmorld) [reply src/components/autofill/autofill.component.js:40]: ahh thanks for the catch and the details with screnshots

It should be possible to pass license number (and possibly other details) and have them applied to the filter of the goTo* method

I'll look into it

**Comment by Sheeri** [comment]
@lenmorld - before you go any further - 2 things:

1 - the code you're submitting fixes a different issue - #62. This code is for the search results page, which is the page you get when you click "See all results for (SEARCH TERM)" as a result of the autocomplete. #62 is the issue for clicking from the autocomplete page.

e.g.  here - see how the URL is #/search_results ? That's the page it refers to. Not your fault at all, I should have put a screenshot in there.
Screen Shot 2020-10-20 at 2 20 00 PM

2 - Someone else is working on #62 and it would be interesting to compare code - see the changes at https://github.com/WildAid/o-fish-web/pull/335/files - which way do you think is better and why? I'm usually of the mindset that simpler is better but sometimes that has unintended consequences, or is a hacky way to do something as opposed to the 'right' way.

**Comment by lenmorld** [comment]
Thanks for clarifying @Sheeri 
I definitely missed the "page" in Search results page 😅 

I think using React router's Link is more idiomatic and declarative since we are using React Router.

It would be good to have one place to manipulate these paths though. `getData` is a good place, instead of one-offing it in the `PreviewItem` component

The sidenote is that we have too many `onClick` -> `goToPage` calls already that is more imperative since it pushes directly to history, which could be wrapped in `<Link>`s instead
We can have a refactor ticket for these to make them Link's instead

image

**Comment by lenmorld** [comment]
...actually, there is one problem with using `<Link>`s
`Link` wraps the item in an `<a>`, so they are styled as such by default: underlined, 
active/visited colors

image

Although it's possible to pass a style, it seems not natural to do so, and there is a risk of a color not being defined in an inner component, thus taking the default colors of an `<a>` tag
Maybe that's why there's a lot of onclick -> goToPage...

What do you think @Sheeri ?

**Comment by lenmorld** [comment]
So for this case, I think it's more consistent to use onClick -> goToPage...
unless there is a better solution to have a no-style `<Link>`

**Comment by sourabhbagrecha** [comment]
> So for this case, I think it's more consistent to use onClick -> goToPage...
> unless there is a better solution to have a no-style `<Link>`
So what onClick does is, it triggers an event when the user clicks on the element, and hence we can not call it a link anymore. 
For navigation, it is better to have Links rather than onClick events because onClick prevents the ability to open in a new tab/window and it is more intuitive & complies with Browser navigation rules. 
Also I would recommend implementing this across the app. Lmk if I am not clear with the explanation.

**Comment by Sheeri** [comment]
@sourabhbagrecha - opening in a new tab/window is a must! however, the link style @lenmorld pointed out is also an issue.

So I think we should go with the "no-style `<Link>`" approach. How does that sound, @lenmorld?

@sourabhbagrecha can you update your PR with the necessary CSS? (or work with @lenmorld and have them update the CSS?)

**Comment by lenmorld** [comment]
@Sheeri @sourabhbagrecha 
From [here](https://stackoverflow.com/questions/37669391/how-to-get-rid-of-underline-for-link-component-of-react-router), we can pass a component to MaterialUI components

```js
<PreviewItem component={Link} to={'/some_link'} />
```

There's also overriding styles, but we might miss some of the other styles
```js
<Link to="first" style={{ textDecoration: 'none' }}>
```
so the first solution might be better, if it works

**Comment by lenmorld** [comment]
Thanks @Sheeri  and @sourabhbagrecha 
I'll update my code here to use the same solution

**Pull request closed without merging**